static/js/vdom.js

```diff
@@ -0,0 +1,132 @@
+exports.eq_array = (a, b, eq) => {
+    if (a === b) {
+        // either both are undefined, or they
+        // are referentially equal
+        return true;
+    }
+
+    if (a === undefined || b === undefined) {
+        return false;
+    }
+
+    if (a.length !== b.length) {
+        return false;
+    }
+
+    return _.all(a, (item, i) => {
+        return eq(item, b[i]);
+    });
+};
+
+exports.ul = (opts) => {
+    return {
+        tag_name: 'ul',
+        opts: opts,
+    };
+};
+
+exports.render = (tag) => {
```

Comment:
Should this be called `render_tag`?  

Also, I think it deserves a security comment that it expects all `attr[1]` values to already be HTML-escaped (or likely better, we should perhaps just escape them?  I think for the use case we have so far, they'll be comma-separated series of ints, but this seems like the kind of thing easily forgotten when reusing this API later)

Reply:
I updated the code to escape them.  Note the commit to move `escape_html` to `util`.  That commit probably makes sense to merge regardless of this patch.

Reply:
Yeah, agreed.